### PR TITLE
add ini option to disable string escape for parametrization

### PR DESCRIPTION
**Pull request by ApaDoctor**
solving https://github.com/pytest-dev/pytest/issues/2482

### CODE REVIEW

File: _pytest/python.py [hunk 1]
```diff
@@ -909,7 +912,11 @@ def _idval(val, argname, idx, idfn, config=None):
             return hook_id
 
     if isinstance(val, STRING_TYPES):
-        return _escape_strings(val)
+        if config is None:
```
Comment (RonnyPfannschmidt) [inline _pytest/python.py]: for sanity and adding a unittest please extract this to a own function

File: _pytest/python.py [hunk 2]
```diff
@@ -909,7 +912,11 @@ def _idval(val, argname, idx, idfn, config=None):
...
+            escape_option = False
+        else:
+            escape_option = config.getini("disable_test_id_escaping_and_forfeit_all_rights_to_community_support")
+        return val if escape_option else _escape_strings(val)
     elif isinstance(val, (float, int, bool, NoneType)):
         return str(val)
     elif isinstance(val, REGEX_TYPE):
```
Comment (RonnyPfannschmidt) [inline _pytest/python.py]: wrt regex its not clear if we should or should not escape

**Comment by RonnyPfannschmidt** [comment]
your re-factoring tool made a mess of the white space bu im inclined to still accept the changes as they make stuff better 👍 

@nicoddemus can we merge master timely to sort out merge issues?

**Comment by nicoddemus** [comment]
@ApaDoctor sorry I hate to ask this, but could you remake the patch without all  the whitespace/formatting changes? I'm sure you can disable that in your editor. Unfortunately as it stands the patch makes a mess of the history because we lose the authors of the changed lines.

@RonnyPfannschmidt definitely about merging this with `master` asap, but I really would rather not having the whitespace changes in place.

**Comment by ApaDoctor** [comment]
Sorry for the long pause. I'll fix whitespace problem in closest time

**Comment by nicoddemus** [comment]
No problem, thanks for the follow up

**Comment by ApaDoctor** [comment]
So i completed all changes. PR can be approved

File: _pytest/python.py [hunk 3]
```diff
@@ -910,6 +914,14 @@ def _find_parametrized_scope(argnames, arg2fixturedefs, indirect):
     return 'function'
 
 
+def _escape_strings_checker(val, config=None):
+    if config is None:
+        escape_option = False
```
Comment (The-Compiler) [inline _pytest/python.py]: Maybe call this `disable_escaping` to make it more clear.

File: doc/en/parametrize.rst [hunk 4]
```diff
@@ -78,6 +78,16 @@ them in turn::
     test_expectation.py:8: AssertionError
     ======= 1 failed, 2 passed in 0.12 seconds ========
 
+
+.. note::
+
+    pytest by default escapes any non-ascii characters used in unicode strings
+    for the parametrization because it has several downsides.
+    If however you would like to use them, use the option disable_test_id_escaping_and_forfeit_all_rights_to_community_support = False
```
Comment (The-Compiler) [inline doc/en/parametrize.rst]: `= True`?

File: doc/en/parametrize.rst [hunk 5]
```diff
@@ -78,6 +78,16 @@ them in turn::
     test_expectation.py:8: AssertionError
     ======= 1 failed, 2 passed in 0.12 seconds ========
 
+
+.. note::
+
+    pytest by default escapes any non-ascii characters used in unicode strings
+    for the parametrization because it has several downsides.
+    If however you would like to use them, use the option disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
```
Comment (nicoddemus) [inline doc/en/parametrize.rst]: Small nit:

```rst
If however you would like to use them, use this option in your ``pytest.ini``:

.. code-block:: ini

    [pytest]
    disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
```

File: _pytest/python.py [hunk 6]
```diff
@@ -82,7 +82,10 @@ def pytest_addoption(parser):
     parser.addini("python_functions", type="args", default=["test", ],
                   help="prefixes or glob names for Python test function and "
                   "method discovery")
-
+    parser.addini("disable_test_id_escaping_and_forfeit_all_rights_to_community_support", type="bool",
+                  default=False, help="disable string escape non-ascii"
+                                      " characters, might cause unwanted side effects")
```
Comment (nicoddemus) [inline _pytest/python.py]: I think it worth adding a "(use at your own risk)" here.

**Comment by nicoddemus** [comment]
Not sure, @ApaDoctor simply closed this, I assumed he wanted to withdraw his contribution... :confused:

**Pull request closed without merging**